tests/sos_tests.py

```diff
@@ -261,6 +262,20 @@ def check_arch_for_enablement(self):
         raise TestSkipError(f"Unsupported architecture {sys_arch} for test "
                             f"(supports: {self.arch})")
 
+    def check_os_version_for_enablement(self):
+        """
+        Check if the test case is meant only for a specific version or versions
+
+        Takes the `versions` class attribute, a list that specifies
+        the versions where the test applies. If the list is empty, assume all
+        versions of the OS are acceptable. Otherwise, raise a TestSkipError.
+        """
+        os_version = distro.detect().version
```

Comment:
As the `distro.detect()` is called on four places and executed many times, does it deserve to set `self.distro = distro.detect()` once and refer to the variable?

Reply:
I think so. I'll wait for other's thoughts and then will push a fix including this change (unless you think a new PR is necessary, but I don't think so?)

Reply:
Agreed, set it once, use it everywhere.